### PR TITLE
Improve the Trust admin view hospital page

### PR DESCRIPTION
**Pull request by jivdhaliwal**
# What
Adds two new panels to the view hospital page displaying the most and least visited Wards in that Hospital. Also updates the table of wards to include number of booked visits and restyled edit / delete ward links

# Why
To give a better performance overview of a Hospital

# Screenshots
ScreenShot 2020-06-04 at 09 57 37

# Notes
This PR also fixes the issue of the GridColumns going full width on smaller viewports for the NumberTile and Panels (which we want to maintain their width)

### CODE REVIEW

File: src/usecases/retrieveHospitalWardVisitTotals.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    (a, b) => b.total_visits - a.total_visits
+  );
+
+  const mostVisitedResult = sortedResults[0];
+  const leastVisitedResult = sortedResults[sortedResults.length - 1];
+
+  const mostVisited = mostVisitedResult
+    ? {
+        wardName: mostVisitedResult.ward_name,
+        total_visits: parseInt(mostVisitedResult.total_visits),
```
Comment (chubberlisk) [inline src/usecases/retrieveHospitalWardVisitTotals.js:32]: I think we should be using camelCase here.

```suggestion
        totalVisits: parseInt(mostVisitedResult.total_visits),
```

File: src/usecases/retrieveHospitalWardVisitTotals.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    [hospitalId]
+  );
+
+  let wards = [];
+
+  queryResult.map(({ ward_id, total_visits }) => {
+    wards[ward_id] = parseInt(total_visits);
+  });
+
+  const sortedResults = queryResult.sort(
```
Comment (chubberlisk) [inline src/usecases/retrieveHospitalWardVisitTotals.js:22]:
```suggestion
  const wardsSortedByTotalVisits = queryResult.sort(
```

File: src/usecases/retrieveHospitalWardVisitTotals.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    (a, b) => b.total_visits - a.total_visits
+  );
+
+  const mostVisitedResult = sortedResults[0];
```
Comment (chubberlisk) [inline src/usecases/retrieveHospitalWardVisitTotals.js:26]:
```suggestion
  const mostVisitedWard = sortedResults[0];
```

Or `wardWithMostVisits`?

File: src/usecases/retrieveHospitalWardVisitTotals.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  const leastVisitedResult = sortedResults[sortedResults.length - 1];
```
Comment (chubberlisk) [inline src/usecases/retrieveHospitalWardVisitTotals.js:27]:
```suggestion
  const leastVisitedWard = sortedResults[sortedResults.length - 1];
```

Or `wardWithLeastVisits`?

File: src/components/WardsTable/index.js [hunk 5]
```diff
@@ -1,58 +1,54 @@
...
         {wards.map((ward) => (
           <tr key={ward.callId} className="nhsuk-table__row">
-            <td className="nhsuk-table__cell">{ward.hospitalName}</td>
+            {!wardVisitTotals && (
+              <td className="nhsuk-table__cell">{ward.hospitalName}</td>
+            )}
             <td className="nhsuk-table__cell">{ward.name}</td>
             <td className="nhsuk-table__cell">{ward.code}</td>
+            {wardVisitTotals && (
+              <td className="nhsuk-table__cell">{wardVisitTotals[ward.id]}</td>
```
Comment (chubberlisk) [inline src/components/WardsTable/index.js:39]:
```suggestion
              <td className="nhsuk-table__cell">{wardVisitTotals[ward.id] || 0 }</td>
```

Or this 0 handled somewhere else as you get this if a ward has no booked visits / no row in the `wardVisitTotals` table.

image

Comment (jivdhaliwal) [reply src/components/WardsTable/index.js:39]: Good point. I'll update the usecase to ensure that wards with no visits are still returned and set the total to 0

**Pull request merged**